pkg/tuf/options.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tuf
+
+import (
+	"embed"
+	"os"
+	"path/filepath"
+)
+
+//go:embed repository
+var embeddedRepo embed.FS
+
+const DefaultMirror = "https://tuf-repo-cdn.sigstore.dev"
+
+// Options represent the various options for a Sigstore TUF Client
+//
+// Note that currently, the cache control is not working. Upon initialization
+// the client will *ALWAYS* perform a TUF update.
+type Options struct {
+	// CacheValidity period in days (default 1)
+	CacheValidity int
+	// ForceCache controls if the cache should be used without update
+	// as long as the metadata is valid
+	ForceCache bool
+	// Root is the TUF trust anchor
+	Root []byte
+	// CachePath is the location on disk for TUF cache
+	// (default $HOME/.sigstore/tuf)
+	CachePath string
+	// RepositoryBaseURL is the TUF repository location URL
+	// (default https://tuf-repo-cdn.sigstore.dev)
+	RepositoryBaseURL string
+	// DisableLocalCache mode allows a client to work on a read-only
+	// files system if this is set, cache path is ignored.
+	DisableLocalCache bool
+}
+
+// DefaultOptions returns an options struct for the public good instance
+func DefaultOptions() (*Options, error) {
```

Comment:
Yeah, this is actually a one. 

The embed operations **does not fail** as long as the correct path is provided, so that removes one case where the function could error.

> If `UserHomeDir()` fails, we could have some default path 

This is the tricky one I believe, as we would need to cover a few OS, which depends on different filesystem layouts. What about init the RUF root in the default temporary directory? This function does not fail. 
See my proposed change.

Reply:
This suggestion: https://github.com/sigstore/sigstore-go/pull/41/files#r1427737332